pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/BaseIndexHandler.java

```diff
@@ -41,13 +41,17 @@
 public abstract class BaseIndexHandler implements IndexHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(BaseIndexHandler.class);
 
-  protected final SegmentMetadata _segmentMetadata;
   protected final IndexLoadingConfig _indexLoadingConfig;
   protected final Set<String> _tmpForwardIndexColumns;
+  protected final SegmentDirectory _segmentDirectory;
 
-  public BaseIndexHandler(SegmentMetadata segmentMetadata, IndexLoadingConfig indexLoadingConfig) {
-    _segmentMetadata = segmentMetadata;
+  // The segmentMetadata may need to be updated after creating the forward index
+  protected SegmentMetadata _segmentMetadata;
```

Comment:
Since this can be modified during the index handling, should we not put it as a member variable, but always read it from the `_segmentDirectory`? The one inside `_segmentDirectory` should always be updated

Reply:
done